packages/mock-account-provider/app/routes/webhooks.ts

```diff
@@ -96,11 +106,19 @@ export async function handleOutgoingPaymentCreated(wh: WebHook) {
   const acc = await mockAccounts.getByPaymentPointer(pp)
 
   if (!acc) {
-    throw json('No account found for payment pointer', { status: 500 })
+    console.log('No account found for payment pointer')
+    return
   }
 
   const amt = parseAmount(payment['sendAmount'])
-  await mockAccounts.pendingDebit(acc.id, amt.value)
+
+  try {
+    await mockAccounts.pendingDebit(acc.id, amt.value)
+  } catch (e) {
```

Comment:
I was going to suggest that we don't have to catch here, since we're not `await`ing `handleOutgoingPaymentCompletedFailed` (we return `200` even if this throws).
But the mock account provider restarts with the uncaught error:
```
local-fynbos-1        | POST /webhooks 200 - - 1.919 ms
local-fynbos-1        | 
local-fynbos-1        | /workspace/packages/mock-account-provider/build/index.js:338
local-fynbos-1        |       throw new Error("invalid pending debit amount, insufficient funds");
local-fynbos-1        |             ^
local-fynbos-1        | Error: invalid pending debit amount, insufficient funds
local-fynbos-1        |     at AccountProvider.pendingDebit (/workspace/packages/mock-account-provider/build/index.js:338:13)
local-fynbos-1        |     at handleOutgoingPaymentCreated (/workspace/packages/mock-account-provider/build/index.js:617:29)
local-fynbos-1        |     at processTicksAndRejections (node:internal/process/task_queues:96:5)
local-fynbos-1        | undefined
local-fynbos-1        | /workspace/packages/mock-account-provider:
local-fynbos-1        |  ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  mock-account-provider@ start: `remix-serve build`
local-fynbos-1        | Exit status 1
local-fynbos-1 exited with code 1
local-fynbos-1        | 
local-fynbos-1        | > mock-account-provider@ start /workspace/packages/mock-account-provider
local-fynbos-1        | > remix-serve build
local-fynbos-1        | 
local-fynbos-1        | calling run_seed
```

Reply:
Doesn't this mean that every action which is going to modify the account's balance should be wrapped in a `try-catch` block, even for `handleOutgoingPaymentCompletedFailed` and `handleIncomingPaymentCompletedExpired` ?

Eg, if calling `mockAccouts.debit` and an error is thrown, this would not restart the MAP too ?

https://github.com/interledger/rafiki/blob/df6311d8a24432c640b876ce8877fbc33efb5903/packages/mock-account-provider/app/routes/webhooks.ts#L87-L90

https://github.com/interledger/rafiki/blob/df6311d8a24432c640b876ce8877fbc33efb5903/packages/mock-account-provider/app/routes/webhooks.ts#L171

Reply:
:thinking: yeah
I was going to suggest we wrap the `switch` block in a `try`/`catch` but then we couldn't respond `400` in the `default`...
https://github.com/interledger/rafiki/blob/73f573d7dcd72eadaac921fc5cbece5a59002fa1/packages/mock-account-provider/app/routes/webhooks.ts#L41-L50

Reply:
~but then we couldn't respond 400 in the default~
Actually, I think that would work (I was thinking `throw` instead of just `return json` for the `400`).
I'm leaning towards having the only one of these :point_down: in a `catch` wrapping the `switch` statement
```
    const errorInfo = parseError(e)
    console.log(errorInfo)
```
That way we're catching all errors, including:
https://github.com/interledger/rafiki/blob/73f573d7dcd72eadaac921fc5cbece5a59002fa1/packages/mock-account-provider/app/routes/webhooks.ts#L126

Then any `json` can be removed from the `handle*` functions. (`throw json` can be converted to `throw new Error`)
Maybe the `handle*` functions could even be moved to a new `/lib/webhooks.server.ts`.